kivy/core/camera/camera_avfoundation_implem.m

```diff
@@ -158,6 +272,52 @@ - (CameraFrame*)getOutput;
     [localpool drain];
 }
 
+bool Camera::attemptFrameRateSelection(int desiredFrameRate){
+    bool isFPSSupported = NO;
+    AVCaptureDeviceFormat *currentFormat = [mCaptureDevice activeFormat];
+    for ( AVFrameRateRange *range in currentFormat.videoSupportedFrameRateRanges ) {
+        if ( range.maxFrameRate >= desiredFrameRate && range.minFrameRate <= desiredFrameRate )        {
+            isFPSSupported = YES;
+            break;
+        }
+    }
+
+    if( isFPSSupported ) {
+        if ( [mCaptureDevice lockForConfiguration:NULL] ) {
+            mCaptureDevice.activeVideoMaxFrameDuration = CMTimeMake( 1, desiredFrameRate );
+            mCaptureDevice.activeVideoMinFrameDuration = CMTimeMake( 1, desiredFrameRate );
+            [mCaptureDevice unlockForConfiguration];
+        }
+    } else {
+        NSLog(@"Selected FPS not available on this platform.");
+    }
+    return isFPSSupported;
+}
+
+bool Camera::attemptCapturePreset(AVCaptureSessionPreset preset){
+    // See available presets: https://developer.apple.com/documentation/avfoundation/avcapturesessionpreset
+    if([mCaptureSession canSetSessionPreset: preset]){
+        [mCaptureSession setSessionPreset: preset];
+        return true;
+    }
+    NSLog(@"Selected preset not available on this platform");
```

Comment:
same, if the present has a useful name to display.

Reply:
Nice catch, added it.